### ISSUE TITLE
Can't import from forumsys.com with 5.0

### ISSUE THREAD

**Issue by roddypratt** [Adldap2/Adldap2-Laravel#614]
- Laravel Version: 5.7
- Adldap2-Laravel Version: 5.0
- PHP Version: 7.1
- LDAP Type: OpenLDAP 

### Description:
Whenever I try an import from ldap.forumsys.com  I get zero users.  Wireshark shows the connection being made and the LDAP server returning zero users. 

My ldap.php has this:-
```
            'settings' => [
                'schema' => Adldap\Schemas\OpenLDAP::class,

                'hosts' => ['ldap.forumsys.com'],
                'port' => 389,
                'timeout' => 5,

                'base_dn' => 'dc=example,dc=com',

                'follow_referrals' => false,
                'use_ssl' => false,
                'use_tls' => false,
            ],
```
Here's the results...
![image](https://user-images.githubusercontent.com/1199520/48132116-4709f800-e28a-11e8-9f00-96ba1553409a.png)

and here's a Wireshark screenshot.
![image](https://user-images.githubusercontent.com/1199520/48132008-f72b3100-e289-11e8-8eb2-dcd61d061c0f.png)

I'm new to adldap2-laravel, so highly likely it's my cockup...

**Comment by roddypratt** [comment]
Just done some more research:-

- adldap2-laravel V4 behaves identically, 
- `dd(Adldap::search()->all());` looks sensible and includes expected data.
- The 'filter' parameter on the search request seems to be the problem.
- Using the `--filter=` option allows me to add to the filter, but not replace it.

So it looks like this is just a restriction with the artisan import command?

**Comment by stevebauman** [comment]
Hi @roddypratt, can you post your `ldap_auth.php` file?

**Comment by roddypratt** [comment]
My `ldap_auth.php` is stock. I haven't modified it.

**Comment by roddypratt** [comment]
Thanks for the pointer. Yes, I hadn't modified the 'scopes' in `ldap_auth.php`.

Would be neat to have that follow the schema from the connection settings. ;)

**Comment by stevebauman** [comment]
Okay hmm... let me try connecting to their server as well on my end and run an import to see if I'm successful!

Also, what do you mean by follow the schema from the connection settings?

**Comment by roddypratt** [comment]
Thanks. I may be confusing myself switching between V4 and V5 here. Once I'd added `Adldap\Laravel\Scopes\UidScope::class` to the V4 adldap_auth.php scopes, I do see records returned. However, with the same change made to V5 ldap_auth.php I still get nothing. 

I've run some tests and the  problem seems to relate to the `(!(objectclass=contact))` part of the filter, which is new in V5. As soon as this is added, nothing comes back from the server - Even though the records don't seem to have a 'contact' objectclass...

As for "follow the schema" I'd wrongly assumed that having selected the OpenLDAP schema in ldap.php that ldap_auth.php would use the correct defaults for ldap_auth.php.  I probably need to read and understand the documentation some more.

**Comment by viniciusraupp** [comment]
Laravel Version: 5.7
Adldap2-Laravel Version: 5.0
PHP Version: 7.1
LDAP Type: ActiveDirectory

I am also having this error after upgrading to version 5.0, I followed all the steps of the documentation for the upgrade. No user is returned.

**Comment by stevebauman** [comment]
Hi @roddypratt, the filter you mentioned was definitely the issue. This was fixed in the core package in [v9.1.4](https://github.com/Adldap2/Adldap2/releases/tag/v9.1.4).

Related: https://github.com/Adldap2/Adldap2-Laravel/issues/637

Can you run a `composer update` and give it another shot? If you still have issues, let me know and I'd be glad to help you debug further, thanks!

**Comment by stevebauman** [comment]
Closing due to inactivity.
